peer/peer.go

```diff
@@ -89,38 +93,99 @@ func New(ip string, port int) (*Peer, error) {
 }
 
 // Receive is the function that gets called when a remote peer
-// opens a new stream with the host that SetStreamHandler() is called on.
-// This should be passed as the second argument to SetStreamHandler().
-// We may want to implement another type of StreamHandler in the future.
+// opens a new stream with this peer.
+// This should be passed as the second argument to SetStreamHandler() after this
+// peer is initialized.
 func (p *Peer) Receive(s net.Stream) {
-	log.Debug("Setting basic stream handler.")
-	p.doCumulus(s)
-}
+	defer s.Close()
```

Comment:
This function will only terminate when we are fully finished communicating with the peer on the other end of the stream. That functionality has yet to be implemented.